gobblin-utility/src/main/java/org/apache/gobblin/util/filesystem/ThrottledFileSystem.java

```diff
@@ -53,13 +53,18 @@
    * Factory for {@link ThrottledFileSystem}.
    */
   public static class Factory<S extends ScopeType<S>> extends FileSystemInstrumentationFactory<S> {
+    private final String SERVICE_NAME_CONF_KEY = "gobblin.broker.limiter.serviceName";
     @Override
     public FileSystem instrumentFileSystem(FileSystem fs, SharedResourcesBroker<S> broker,
         ConfigView<S, FileSystemKey> config) {
       try {
-        Limiter limiter =
-            broker.getSharedResource(new SharedLimiterFactory<S>(), new FileSystemLimiterKey(config.getKey().getUri()));
-        return new ThrottledFileSystem(fs, limiter);
+        String serviceName = config.getConfig().getString(SERVICE_NAME_CONF_KEY);
+        Limiter limiter;
+        if (serviceName == null)
+          limiter = broker.getSharedResource(new SharedLimiterFactory<S>(), new FileSystemLimiterKey(config.getKey().getUri()));
```

Comment:
No need for this `if` check, you can use the same constructor, it will just get a `null` service name in some cases.